packages/roosterjs-content-model-plugins/lib/touch/repositionTouchSelection.ts

```diff
@@ -0,0 +1,84 @@
+import type { IEditor, ContentModelText } from 'roosterjs-content-model-types';
+import { getSelectedSegmentsAndParagraphs } from 'roosterjs-content-model-dom';
+import { adjustWordSelection } from 'roosterjs-content-model-api';
+
+const MAX_TOUCH_MOVE_DISTANCE = 6; // the max number of offsets for the touch selection to move
+
+/**
+ * @internal
+ * Touch selection, if tap within 6 characters of the beginning/end. Find the nearest edge of the word and move cursor there.
+ */
+export function repositionTouchSelection(editor: IEditor) {
+    editor.formatContentModel(
+        (model, _context) => {
```

Comment:
You can extract this part as a function and call formatContentModel from the plugin, so it will be easier to add unit test to this function.